syndeo-template/TESTING.md

```diff
@@ -0,0 +1,81 @@
+# Testing of the `syndeo-template` package
+
+## Testing the Kafka-to-BigQuery solution
+
+The end-to-end test for the Kafka-to-BigQuery solution is implemented in `KafkaToBigQueryIT#testKickOffKafkaToBigQuerySyndeoPipeline`.
+The test performs the following set of actions:
+
+1. Start a pipeline that generates random data and writes it to a Kafka cluster as Avro-encoded records (data generator)
+   1. **Note**: This pipeline generates a fixed number of records.
+2. Start a pipelnie from the Syndeo template that will consume data from the Kafka cluster (pipeline #1)
+3. Let the pipeline run for a short time
+4. Drain pipeline #1
+5. Start a new pipeline with the Syndeo template with the same configuration as #1 (this new pipeline is pipeline #2)
+6. Query BigQuery until it contains at least the expected number of records.
+7. Terminate the second Syndeo pipeline
+8. (implicit) The data generator will terminate on its own after generating the fixed number of elements.
+
+After performing those actions, the test then verifies that only the expected number of records were written to BigQuery
+(no data was duplicated.)
+
+The end-to-end test for the Kafka-to-BigQuery solution assumes various pieces of infrastructure exist, and relies on
+them to run. The infrastructure that the test expects is:
+
+- A locally set-up `DT_IT_ACCESS_TOKEN` variable that allows the testing framework to authenticate to GCP.
```

Comment:
If the access token is not set it uses application default credentials (which are set up using `gcloud auth application-default login`

Reply:
done